source/tests/unit/xnet_converters_tests.cpp

```diff
@@ -104,6 +104,191 @@ TEST(XnetConvertersTests, SessionInfoStateValue_SetSessionInfoResponse_ExtractSe
+}
+
+u32 GenerateVlanTag(u16 vlanId)
+{
+  u32 vlanTag = 0x81000000;
+  u8 kPcp = 3;
+  AssignVlanId(&vlanTag, vlanId);
+  AssignPriorityCodePoint(&vlanTag, kPcp);
+  return vlanTag;
+}
+
+void EncodeEnetFrameHeader(nxFrameEnet_t* pFrame, bool hasVlan, u16 vlanId)
+{
+  struct EthernetHeader EnetHead =
+      {
+          {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+          {0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF},
+          0x88B5 /* Local Experimental Ethertype */
+      };
+
+  u8 kEnetFrameOffsetEtherType = 12;
+  u8 kEnetFrameOffsetVlanEtherType = 16;
+  u8 kEnetFrameOffsetVlanTag = 12;
+  u8 kEnetFrameOffsetDstMacAddr = 0;
+  u8 kEnetFrameOffsetSrcMacAddr = 6;
+  u32 kEnetMacAddrSize = 6;
+
+  u16 ethertypeBigEndian = ChangeByteOrderU16(EnetHead.EtherType);
+  u32 vlanTagBigEndian = ChangeByteOrderU32(GenerateVlanTag(vlanId));
+
+  memcpy(pFrame->FrameData + kEnetFrameOffsetDstMacAddr, &EnetHead.DestinationMacAddress, kEnetMacAddrSize);
+  memcpy(pFrame->FrameData + kEnetFrameOffsetSrcMacAddr, &EnetHead.SourceMacAddress, kEnetMacAddrSize);
+
+  if (hasVlan) {
+    memcpy(pFrame->FrameData + kEnetFrameOffsetVlanTag, &vlanTagBigEndian, sizeof(vlanTagBigEndian));
+    memcpy(pFrame->FrameData + kEnetFrameOffsetVlanEtherType, &ethertypeBigEndian, sizeof(EthernetHeader.EtherType));
+  }
+  else {
+    memcpy(pFrame->FrameData + kEnetFrameOffsetEtherType, &ethertypeBigEndian, sizeof(EthernetHeader.EtherType));
+  }
+}
+
+void EncodeEnetFrame(nxFrameEnet_t* pFrame, u16 payloadSize, bool hasVlan, u16 vlanId)
+{
+  u16 frameSize = 0;
+  u16 frameSizeWithoutPayload = sizeof(nxFrameEnet_t) - 1;
+
+  u32 kFcsSize = 4;
+  u32 payloadOffset = 0;
+  u32 kEnetFrameHeaderSize = 14;
+  u32 kEnetFrameVlanTaggedHeaderSize = 18;
+
+  EncodeEnetFrameHeader(pFrame, hasVlan, vlanId);
+  pFrame->DeviceTimestamp = 0;
+  pFrame->NetworkTimestamp = 0;
+  pFrame->Flags = nxEnetFlags_Transmit;
+  pFrame->Type = nxEnetFrameType_Data;
+
+  if (hasVlan) {
+    payloadOffset = kEnetFrameVlanTaggedHeaderSize;
+  }
+  else {
+    payloadOffset = kEnetFrameHeaderSize;
+  }
+
+  frameSize = (u16)(frameSizeWithoutPayload + payloadOffset + payloadSize + kFcsSize);
+  pFrame->Length = frameSize;
+
+  for (u16 i = 0; i < payloadSize; i++) {
+    pFrame->FrameData[payloadOffset + i] = (u8)i;
+  }
+}
+
+void assert_enet_frames_are_equal(nxFrameEnet_t* frame1, nixnet_grpc::EnetFrameResponse frame2)
+{
+  EXPECT_EQ(nixnet_grpc::EnetFrameType::ENET_FRAME_TYPE_DATA, frame2.type());
+  EXPECT_EQ(frame1->Type, frame2.type_raw());
+  EXPECT_EQ(frame1->DeviceTimestamp, frame2.device_timestamp());
+  EXPECT_EQ(frame1->NetworkTimestamp, frame2.network_timestamp());
+  EXPECT_EQ(nixnet_grpc::EnetFlags::ENET_FLAGS_TRANSMIT, frame2.flags_mapped()[0]);
+  EXPECT_EQ(frame1->Flags, frame2.flags_raw());
+  ASSERT_THAT(std::vector<u8>(frame1->FrameData, frame1->FrameData + frame1->Length - sizeof(nxFrameEnet_t) + 1), ::testing::ElementsAreArray(frame2.frame_data()));
```

Comment:
Add a comparison for `frame1->Length - nixnet_grpc::ENET_FRAME_HEADER_LENGTH` and `frame2.frame_data().size()`.

@Akshata440 once you fix enet frame bug, you will need to fix above check to substract `FCS_SIZE` from `frame->Length` and also `ASSERT_THAT` call here should compare 4 bytes less.